vscode-dotnet-runtime-library/src/Acquisition/InstallTrackerSingleton.ts

```diff
@@ -185,7 +182,8 @@ Installs: ${[...this.inProgressInstalls].map(x => x.dotnetInstall.installId).joi
      */
     public async getExistingInstalls(getAlreadyInstalledVersion : boolean, alreadyHoldingLock = false) : Promise<InstallRecord[]>
     {
-        return this.executeWithLock( alreadyHoldingLock, (getAlreadyInstalledVersions : boolean) =>
+        return this.executeWithLock( alreadyHoldingLock, getAlreadyInstalledVersion ? this.installedVersionsId : this.installingVersionsId,
+            (getAlreadyInstalledVersions : boolean) =>
```

Comment:
nit: passing in string data key here would be better as the outer function will calculate it based on bool value.

Reply:
I think that's a good point but this interface is also used in a public fashion and I don't want outer callers to be aware of the string data key. If you disagree, please let me know.